google-cloud-logging/lib/google/cloud/logging/project.rb

```diff
@@ -332,11 +332,12 @@ def async_writer max_queue_size: AsyncWriter::DEFAULT_MAX_QUEUE_SIZE
         #   resource to be associated with written log entries.
         # @param [Hash] labels A set of user-defined data to be associated with
         #   written log entries.
-        # @param [Boolean|AsyncWriter] async_writer An AsyncWriter for the
-        #   logger to transmit log entries. You may also pass true to request
-        #   a new AsyncWriter for this logger, or false to request no
-        #   AsyncWriter (which will cause the logger to make blocking calls).
-        #   Default is true.
+        # @param [Boolean] async Whether entries are written asynchronously.
+        #   When `true` the logger will use use a new AsyncWriter instance. When
+        #   `false` it will write entries synchronously using
+        #   Project#write_entries. Default is true.
+        # @param [AsyncWriter|#write_entries] writer The object to write entries
```

Comment:
It might help me if we take a step back then. I have two general questions:

1. What is the use case for providing a writer object? I can think of reasons why it might be handy to have, but none of them seem to be general enough to be part of the public API. What are we asking users to do with this option?
2. What is the use case to allow the Logger to be synchronous? As I said before, I think asynchonous should just be the behavior of Logger, so do we need the flag to turn that off?

I agree that arguments are best when they are orthogonal. But I would vastly prefer separate `#logger` and `#async_logger` methods to an argument that is both a boolean (or a symbol) and a writer object.